### PR TITLE
Add Secret commands

### LINKED ISSUE #907: command to add a secret to a worker
wrangler should support a command that allows the user to securely add a secret environment variable to the script of their choosing. should be interactive by default to prevent sensitive data from leaking into terminal history. ideally also allows for values to be piped to it.

### Syntax
```
$ wrangler secret set secret-name
=> Enter your secret value
# or
$ echo $SECRET_VALUE | wrangler secret secret-name
```
this command should support the environment flag; when not present, the command should create a secret in the context of the top level worker. when present, the command should create a secret in the context of the worker specified in the env.

#### outstanding questions:

1. what to do when the worker does not exist yet
2. does this require extra encryption before sending?

### Depends on
* add endpoint to workers configuration api to add a secret binding to an already existing worker
* add endpoint definition to cloudflare-rs for calling the above endpoint

### PR DESCRIPTION
**Pull request by ashleymichal**
closes #907, #912, #909 . just a repackaging of #994

### CODE REVIEW

File: src/terminal/interactive.rs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+// For interactively handling  reading in a string
+pub fn get_string(prompt_string: &str) -> String {
+    println!("{}", prompt_string);
+    let input: String = read!("{}\n");
```
Comment (EverlastingBugstopper) [inline src/terminal/interactive.rs]: nit: i wonder if you can `read!("{} {}", prompt_string);

File: src/terminal/interactive.rs [hunk 2]
```diff
@@ -0,0 +1,26 @@
+// For interactively handling  reading in a string
+pub fn get_string(prompt_string: &str) -> String {
```
Comment (EverlastingBugstopper) [inline src/terminal/interactive.rs]: nit: rename to `get_user_input`

File: src/terminal/interactive.rs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+const YES: &str = "y";
+const NO: &str = "n";
+// For interactively handling deletes (and discouraging accidental deletes).
+// Input like "yes", "Yes", "no", "No" will be accepted, thanks to the whitespace-stripping
+// and lowercasing logic below.
+pub fn delete(prompt_string: &str) -> Result<bool, failure::Error> {
+    println!("{} [y/n]", prompt_string);
+    let mut response: String = read!("{}\n");
+    response = response.split_whitespace().collect(); // remove whitespace
+    response.make_ascii_lowercase(); // ensure response is all lowercase
```
Comment (EverlastingBugstopper) [inline src/terminal/interactive.rs]: is there a reason we use `make_ascii_lowercase` instead of `to_lowercase`?

File: src/commands/secret/mod.rs [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+}
+
+// secret_errors() provides more detailed explanations of Workers KV API error codes.
+// See https://api.cloudflare.com/#workers-secrets ? for details.
+fn secret_errors(error_code: u16) -> &'static str {
+    match error_code {
+        7003 | 7000 => {
+            "Your wrangler.toml is likely missing the field \"account_id\", which is required to write to Workers KV."
+        }
+        10001 => "The content you passed in is not an accepted string. Try entering just a string",
```
[the remark this answers is not in the document]
Comment (ashleymichal) [reply src/commands/secret/mod.rs]: i think this is not a thing that gets returned for these endpoints anyway.. will remove.

File: src/commands/secret/mod.rs [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+        10053 => "There is already another binding with a different type by this name. Check your wrangler.toml or dashboard for conflicting bindings",
```
[the remark this answers is not in the document]
Comment (ashleymichal) [reply src/commands/secret/mod.rs]: so this happens in the  case that there is a binding by the same name but of another kind, so perhaps a kv namespace or plain text binding, for example. this isn't in the case of a secret by the same name, because this is a set so it'll just overwrite.

File: src/terminal/interactive.rs (hunk 1 above)
Comment (ashleymichal) [reply src/terminal/interactive.rs]: it got mad, or i don't understand.

File: src/terminal/interactive.rs (hunk 3 above)
Comment (ashleymichal) [reply src/terminal/interactive.rs]: idk, @victoriabernard92 ?

Comment (victoriabernard92) [reply src/terminal/interactive.rs]: This was copy from gabbi's work on interactive delete

Comment (ashleymichal) [reply src/terminal/interactive.rs]: interesting. it is oddly specific...perhaps we should leave it?

Comment (ashleymichal) [reply src/terminal/interactive.rs]: i'm going to revert it to using this, and cc @gabbifish for when she gets back to see if we can change it. if not, i'd like to add a comment so we don't re-litigate this in the future.

File: src/commands/secret/mod.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (EverlastingBugstopper) [reply src/commands/secret/mod.rs]: ah. how about `"There is already a KV namespace or text binding with this name associated with your script"`or something like that/ "binding with a different type by this name" is hard to parse

[the remark this answers is not in the document]
Comment (ashleymichal) [reply src/commands/secret/mod.rs]: there is a world in which it is actually a text blob binding you are using for sites, or you went and called it `wasm` or something else that's taken by our build process. if we wanted to be really robust, we could query the list of bindings to show what's up, check their toml, look to see if they're stomping  on their own asset manifest or wasm binding... this feels like a good add-on issue though.

[the remark this answers is not in the document]
Comment (EverlastingBugstopper) [reply src/commands/secret/mod.rs]: 👍
